src/main/java/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,39 @@
+package cinema.dao.impl;
+
+import cinema.dao.MovieDao;
+import cinema.exception.DataProcessingException;
+import cinema.lib.Dao;
+import cinema.model.Movie;
+import cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
According to the checklist we need to use try-finally and close session manually